### PR TITLE
Search for user via previous username

### LINKED ISSUE #4471: Search (and redirect) for users via previous usernames
Until somebody else claims a username, old usernames should easily redirect to the user who had them previously.
For example, if you put /users/oldname , it should redirect to /users/newname , or if you put oldname in the user search, newname should be a result. (maybe with something added to the user card, like was: oldname in small font to make clear why they are a result)

This would be to make finding users even after they change names easier.
It is especially relevant when looking at maps who say "GD by blah" but you can't find who blah is, because the mapper didn't link the userpage by id number, and blah changed their name!

URL redirects should be so long as nobody else takes the old nickname. The search results should be permanent except for cases where support manually removes names from a user's name history.

### PR DESCRIPTION
**Pull request by notbakaneko**
A new elasticsearch index to replace the old one needs to be created, or the fields manually added.

continues from #4546
closes #4471

### CODE REVIEW

File: app/Console/Commands/EsIndexDocuments.php [hunk 1]
```diff
@@ -144,13 +148,70 @@ private function indexGroup($name)
...
+
         if ($alias !== $indexName) {
             $this->info("Aliasing {$alias} to {$indexName}");
             Indexing::updateAlias($alias, [$indexName]);
             $this->line('');
         }
     }
 
+    // TODO: should go somewhere else
+    protected function afterIndexUsers($indexName)
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply app/Console/Commands/EsIndexDocuments.php]: Any extra query per individual user during bulk indexing should be avoided.

[the remark this answers is not in the document]
Comment (nanaya) [reply app/Console/Commands/EsIndexDocuments.php]: More like ~1600s vs ~1400s on my pc.

[the remark this answers is not in the document]
Comment (notbakaneko) [reply app/Console/Commands/EsIndexDocuments.php]: ~2600s -> ~3500s

File: app/Console/Commands/EsIndexDocuments.php [hunk 2]
```diff
@@ -64,6 +64,8 @@ class EsIndexDocuments extends Command
      */
     public function handle()
     {
+        config()->set('indexing.batch', true);
```
Comment (nanaya) [inline app/Console/Commands/EsIndexDocuments.php]: not used?

File: app/Models/User.php [hunk 3]
```diff
@@ -1597,6 +1609,24 @@ public function playCount()
...
+     * @param bool $includeCurrent true if previous usernames matching the the current one should be included.
+     *
+     * @return \Illuminate\Database\Eloquent\Collection string
+     */
+    public function previousUsernames(bool $includeCurrent = false)
+    {
+        $query = $this->usernameChangeHistoryPublic;
+
+        if (!$includeCurrent) {
+            $query->where('username_last', '<>', $this->username);
```
Comment (nanaya) [inline app/Models/User.php]: should be `$query = $query->where()` as it's now collection. And maybe rename the variable.

**Pull request merged**